presto-main/src/main/java/com/facebook/presto/type/DoubleOperators.java

```diff
@@ -185,6 +185,13 @@ public static long castToLong(@SqlType(StandardTypes.DOUBLE) double value)
     }
 
     @ScalarOperator(CAST)
+    @SqlType(StandardTypes.FLOAT)
+    public static long castToFloat(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return Float.floatToRawIntBits((float) value);
```

Comment:
(my knowledge of approximate types is limited) but does this work for all values of Double?